engine/src/main/java/org/hibernate/search/backend/impl/lucene/MultiWriteDrainableLinkedList.java

```diff
@@ -37,17 +37,39 @@
 	 */
 	public void add(T element) {
 		final Node<T> newnode = new Node<T>( element );
-		addNode( newnode );
+		addNode( newnode, newnode );
 	}
 
-	private synchronized void addNode(Node<T> newnode) {
+	/**
+	 * Adds multiple new entries to this list, in the given order,
+	 * making sure no entry is added between the given elements.
+	 *
+	 * @param elements The elements to add.
+	 */
+	public void addAll(Iterable<T> elements) {
```

Comment:
I was trying to avoid instantiating an object for each set of works (which could be one object _per work_ in the case of `org.hibernate.search.indexes.spi.IndexManager.performStreamOperation(LuceneWork, IndexingMonitor, boolean)`), but in retrospect it seems a bit stupid given that this forces us to instantiate dozens of nodes in the non-stream case... I'll change that.

On a related note, the `monitor` object in `Changeset` is never used. There may be something wrong there... ? I'll remove it.